torch/nn/parallel/distributed.py

```diff
@@ -856,6 +857,7 @@ def forward(self, *inputs, **kwargs):
             # This should be called only once during whole training period.
             if torch.is_grad_enabled() and self.reducer._rebuild_buckets():
                 logging.info("Reducer buckets have been rebuilt in this iteration.")
+                self._has_rebuilt_buckets = True
```

Comment:
Note that this might not be entirely accurate. 

If find_unused_params=True then we don't actually rebuilt buckets and it's basically a noop (you can confirm by seeing reducer code). Not sure if it really affects your use case though.

Reply:
I see. The idea is that I just need a signal that buckets have been finalized since the design I preferred was to only fully initialize the `ZeroRedundancyOptimizer` instance after that point. This made sense to me given that the current DDP implementation rebuilds buckets early and never again, so any initial construction that seems wasteful, as it will soon be thrown away.